pkg/ottl/expression.go

```diff
@@ -117,6 +122,52 @@ func (g StandardTypeGetter[K, T]) Get(ctx context.Context, tCtx K) (T, error) {
 	return v, nil
 }
 
+type StringLikeGetter[K any] interface {
```

Comment:
Would it make sense to use this in the `concat` Converter? I recall we aren't currently converting maps or slices due to the potential to blow up the payload size, but I wonder if we should revisit that decision now that we're standardizing string conversion.

Reply:
Ya I think that makes sense.  `IsMatch` and `Concat` handle []byte different, though, so we need to pick a standard.  This PR currently uses IsMatch's `base64.StdEncoding.EncodeToString(v)` solution instead of Concat's `fmt.Sprintf("%x", v)` solution.

Reply:
Kind feel like both are wrong and we should be using `hex.EncodeToString` like pdata does.

Reply:
Final important difference, Concat ignores values it doesn't understand.  StringLikeGetter will return an error which the function will return.

Reply:
>This PR currently uses IsMatch's base64.StdEncoding.EncodeToString(v) solution instead of Concat's fmt.Sprintf("%x", v) solution. [...] Kind feel like both are wrong and we should be using hex.EncodeToString like pdata does.

Good catch, using `hex.EncodeToString` should be the right way to go.

>Final important difference, Concat ignores values it doesn't understand. StringLikeGetter will return an error which the function will return.

I think that throwing an error is probably the desired behavior, since it provides more consistency around how values are stringified and when they can't be. Realistically we should be capable of providing a string representation for most (if not all) OTTL values.